pymeasure/adapters/adapter.py

```diff
@@ -38,9 +39,13 @@ class Adapter:
     :param kwargs: all other keyword arguments are ignored.
     """
 
+    connection_lock_initialized = False
```

Comment:
I'm wondering: why do you need this separate sentinel attribute that is effectively always true? 
You are creating the lock when initialising the `Adapter` - is it not enough to do that as the first action in `__init__`, and then just use the now existing lock? We would only need to make sure that all subclasses call the superclass init first thing.

To be a bit more defensive, you could set `self.connection_lock = None` here at class level, and check with `is None` (or just `and self.connection_lock`) in the attribute methods, or does that not work?